src/main/ruby/core/regexp.rb

```diff
@@ -384,7 +384,11 @@ def inspect
       "#<MatchData \"#{self[0]}\">"
     else
       idx = 0
-      capts.map! { |capture| "#{idx += 1}:#{capture.inspect}" }
+      capts.map! do |capture|
+        idx += 1
+        name = names[idx - 1] || idx
```

Comment:
It would be a better to only get the Regexp#names once here.
Something like `names = named()` above the `map!` would achieve that.

Reply:
Do we want @tjoyal to change that then? We need to be clear what we're asking for a first-time contributor.

Reply:
Yes, please change it, @tjoyal, to avoid making too many calls and allocations in Regexp#inspect.

Reply:
Just to be clear, names is the method we want to only call once so it'll be `regex_names = names`?

Not a fan of the `name = regex_names[idx - 1] || idx` logic but if we want to focus on allocations it's the best I have.

Reply:
I made a typo, I meant:
```ruby
names = names()
capts.map! do |capture|
  ...
```
So the code below can remain unchanged.
`regex_names = names` instead of `names = names()` is fine too of course.

Reply:
If you want nicer logic, then maybe `capts.zip(names)` would be a nice way to express it.

Reply:
> So the code below can remain unchanged.

I'm good either and will go with your preference, that "name" code is all new. (but I personally prefer to never shadow references, makes moving code later on simpler). 

`capts.zip(names)` wouldn't work out of the box as `names` is an empty array in the cases of unnamed regex captures. What I wanted to avoid is the `||` inlined in the method, but I can't do it without adding some extra logic.

I somewhat feels when Regex capture name is not specified, it could just default the name to the position at that time, but that would not be a change compatible with other implementations.

Reply:
> I'm good either and will go with your preference, that "name" code is all new. (but I personally prefer to never shadow references, makes moving code later on simpler).

Let's go with `regex_names` then.

> `capts.zip(names)` wouldn't work out of the box

`zip` would require more changes, but note that it would handle an empty Array nicely:
```ruby
[1] pry(main)> [1,2,3].zip([])
=> [[1, nil], [2, nil], [3, nil]]
```
So something like (untested):
```ruby
  def inspect
    str = "#<MatchData \"#{self[0]}\""
    idx = 0
    captures.zip(names) do |capture, name|
      idx += 1
      str << " #{name || idx}:#{capture.inspect}"
    end
    "#{str}>"
  end
```
which seems much nicer to me.
But that's a bigger refactoring, so there is no need to do it in this PR if you prefer to keep changes minimal.

Reply:
Ok, I'm still not 100% convinced by that `zip`, but removing the `if` and re-using the same String makes so much more sense. Pushed the change.